owncloudApp/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.kt

```diff
@@ -215,50 +216,14 @@ class FileDataStorageManager : KoinComponent {
 //            return result.apply { sort() }
         }
 
-    val sharedByLinkFilesFromCurrentAccount: Vector<OCFile>
-        get() {
-            return Vector()
-            // FIXME: 13/10/2020 : New_arch: Shared By Link
-//            val allSharedFiles = Vector<OCFile>()
-//            val result = Vector<OCFile>()
-//            var cursorOnShared: Cursor? = null
-//            try {
-//                cursorOnShared = contentResolver?.query(
-//                    CONTENT_URI,
-//                    null,
-//                    "$FILE_SHARED_VIA_LINK = ? AND $FILE_ACCOUNT_OWNER = ? ",
-//                    arrayOf(1.toString(), account.name),
-//                    null
-//                )
-//                if (cursorOnShared != null && cursorOnShared.moveToFirst()) {
-//                    var file: OCFile?
-//                    do {
-//                        file = createFileInstance(cursorOnShared)
-//                        allSharedFiles.add(file)
-//                    } while (cursorOnShared.moveToNext())
-//                }
-//            } catch (exception: Exception) {
-//                Timber.e(exception, "Exception retrieving all the shared by link files")
-//            } finally {
-//                cursorOnShared?.close()
-//            }
-//
-//            if (allSharedFiles.isNotEmpty()) {
-//                val allSharedDirs = Vector<Long>()
-//                for (file in allSharedFiles) {
-//                    if (file.isFolder) {
-//                        allSharedDirs.add(file.fileId)
-//                    }
-//                }
-//                for (file in allSharedFiles) {
-//                    if (file.isFolder || (!file.isFolder && !allSharedDirs.contains(file.parentId))) {
-//                        result.add(file)
-//                    }
-//                }
-//            }
-//            result.sort()
-//            return result
-        }
+    fun sharedByLinkFilesFromCurrentAccount(): List<OCFile>? = runBlocking(CoroutinesDispatcherProvider().io) {
+        val getFilesSharedByLinkUseCase: GetFilesSharedByLinkUseCase by inject()
+
+        val result = withContext(CoroutineScope(CoroutinesDispatcherProvider().io).coroutineContext) {
+            getFilesSharedByLinkUseCase.execute(GetFilesSharedByLinkUseCase.Params(account.name))
+        }.getDataOrNull()
+        result
```

Comment:
I think that we should return an empty list in case there are no shared by link files. What do you think?

Reply:
I agree with you.